arbitrator/prover/src/machine.rs

```diff
@@ -2483,9 +2517,24 @@ impl Machine {
                 hash_stack_with_heights(frames, &heights, concat!($prefix, " stack:"))
             }};
         }
+
+        // compute_slices works similarly as compute, except for vector of slices.
+        // Instead of constructing vector of each Value's hash, it constructs
+        // hash of each slice (&[Value]). Rest is the same.
+        macro_rules! compute_vec {
+            ($field:expr, $stacks:expr, $prefix:expr) => {{
+                let heights: Vec<_> = self.guards.iter().map($field).collect();
+                
+                // Map each &[Value] slice to its hash using `hash_stack` and collect these hashes into a vector
+                let slice_hashes: Vec<_> = $stacks.iter()
```

Comment:
this is the wrong order. This is [0..n-1]  we want [1..n-1, 0]

Reply:
Done.